tools/memap.py

```diff
@@ -389,67 +409,56 @@ def generate_output(self, export_format, file_output=None):
                 k:self.modules[i][k] for k in self.print_sections}})
 
             table.add_row(row)
+        json_obj.append({\
+                'summary':{\
+                'static_ram':(subtotal['.data']+subtotal['.bss']),\
+                'heap':(subtotal['.heap']),\
+                'stack':(subtotal['.stack']),\
+                'total_ram':(subtotal['.data']+subtotal['.bss']+subtotal['.heap']+subtotal['.stack']),\
+                'total_flash':(subtotal['.text']+subtotal['.data']+misc_flash_mem),}})
+
+        file_desc.write(json.dumps(json_obj, indent=4))
+        file_desc.write('\n')
+
+    def export_csv_ci(self, file_desc, _, subtotal, misc_flash_mem):
+        csv_writer = csv.writer(file_desc, delimiter=',', quoting=csv.QUOTE_NONE)
+
+        csv_module_section = []
+        csv_sizes = []
+        for i in sorted(self.modules):
+            for k in self.print_sections:
+                csv_module_section += [i+k]
+                csv_sizes += [self.modules[i][k]]
 
-        subtotal_row = ['Subtotals']
-        for k in self.print_sections:
-            subtotal_row.append(subtotal[k])
-
-        table.add_row(subtotal_row)
-
-        if export_format == 'json':
-            json_obj.append({\
-                  'summary':{\
-                  'static_ram':(subtotal['.data']+subtotal['.bss']),\
-                  'heap':(subtotal['.heap']),\
-                  'stack':(subtotal['.stack']),\
-                  'total_ram':(subtotal['.data']+subtotal['.bss']+subtotal['.heap']+subtotal['.stack']),\
-                  'total_flash':(subtotal['.text']+subtotal['.data']+misc_flash_mem),}})
-
-            file_desc.write(json.dumps(json_obj, indent=4))
-            file_desc.write('\n')
-
-        elif export_format == 'csv-ci': # CSV format for the CI system
-
-            csv_writer = csv.writer(file_desc, delimiter=',', quoting=csv.QUOTE_NONE)
-
-            csv_module_section = []
-            csv_sizes = []
-            for i in sorted(self.modules):
-                for k in self.print_sections:
-                    csv_module_section += [i+k]
-                    csv_sizes += [self.modules[i][k]]
```

Comment:
This look like pretty significant changes to `memap.py`? Should this be in another PR?